### PR TITLE
Refine Frontend Stack

### PR DESCRIPTION
**Pull request by nerd-feller**

### CODE REVIEW

File: frontend-js.md [hunk 1]
```diff
@@ -50,9 +50,14 @@
 
 # Build Tools (Bundlers)
 Module Bundling, on a high level, is a process of integrating together a group of modules in a single file so that multiple modules can be sent to the browser in a single bundle (which is good for performance but bad for debugging).
-- [Webpack](https://webpack.js.org) - a modular build tool 
+- [Webpack](https://webpack.js.org) - a modular build tool
 - [Parcel](https://parceljs.org/) - gives you “blazing fast” bundling as it uses multiple worker processes to ensure that the compilation process is executed in parallel on multiple cores without the need of any configuration
 
+# Module Loader
```
Comment (lankastersky) [inline frontend-js.md]: May be move it under 'Libraries' section?
And please add the link to the TOC on the top

File: frontend-js.md [hunk 2]
```diff
@@ -91,16 +97,24 @@ Module Bundling, on a high level, is a process of integrating together a group o
 - [karma-htmlfile-reporter](https://github.com/matthias-schuetz/karma-htmlfile-reporter) - a karma plugin for exporting unit test results as styled HTML file
 - [karma-spec-reporter](https://github.com/mlex/karma-spec-reporter#readme) - Test reporter that prints detailed results to console (similar to mocha's spec reporter).
 
+### Task runner
```
Comment (lankastersky) [inline frontend-js.md]: why is it in testing frameworks? 
Can you add a short description of what is it?
Please update TOC

File: frontend-js.md [hunk 3]
```diff
@@ -91,16 +97,24 @@ Module Bundling, on a high level, is a process of integrating together a group o
...
+- [Grunt](https://gruntjs.com/)
+
 ## Transpilers
 Transpilers - Source-to-source compilers, tools that read source code written in one programming language, and produce the equivalent code in another language
 - [Babel](https://babeljs.io) - converts new JS code into old ones
+- [TypeScript](https://www.typescriptlang.org/index.html) - TypeScript is a superset of JavaScript that compiles to plain JavaScript.
 
-## CSS libraries
-- Flexbox - new layout mode in CSS3
+## CSS preprocessor
```
Comment (lankastersky) [inline frontend-js.md:111]: Please update TOC

File: frontend-js.md [hunk 4]
```diff
@@ -91,16 +97,24 @@ Module Bundling, on a high level, is a process of integrating together a group o
...
 - [Sass](https://sass-lang.com/) - CSS pre-processor with syntax advancements (older syntax)
 - Scss - CSS extension, preprocessor which lets you use features that aren’t a part of the wider CSS standard yet, and provides better workflows for maintaining your stylesheets
 - [Stylus](http://stylus-lang.com/) - Expressive, robust, feature-rich CSS language built for nodejs
 
+## CSS Layout
```
Comment (lankastersky) [inline frontend-js.md:117]: please update TOC

File: frontend-js.md (hunk 1 above)
Comment (nerd-feller) [reply frontend-js.md]: Done

File: frontend-js.md (hunk 2 above)
Comment (nerd-feller) [reply frontend-js.md]: Fixed

File: frontend-js.md (hunk 3 above)
Comment (nerd-feller) [reply frontend-js.md:111]: Done

File: frontend-js.md (hunk 4 above)
Comment (nerd-feller) [reply frontend-js.md:117]: Done

**Pull request merged**